src/OpenTelemetry.Api/Trace/SpanCreationOptions.cs

```diff
@@ -38,7 +38,7 @@ public class SpanCreationOptions
         /// <summary>
         /// Gets or sets attributes known prior to span creation.
         /// </summary>
-        public IDictionary<string, object> Attributes { get; set; }
+        public IEnumerable<KeyValuePair<string, object>> Attributes { get; set; }
```

Comment:
As far as I can tell, IDictionary<string, object> implements IEnumerable<KeyValuePair,string, object>> so we are just making it more generic. From the issue, I understood that we want to maintain sequence/order. I believe IEnumerable<> won't fix the issue.

Reply:
IEnumerable should be enough and the callers or the creators of the collection can decide how to create it. If the caller cares about the order, then they will need to create it using something like `List<KeyValuePair<string, object>>`. If the caller don't care about order they can create it using a dictionary. From the API perspective, we should accept `IEnumerable<KeyValuePair<string, object>>` which maintains the order after that as the specs require. I agree with OT API/SDK implementation shouldn't use IDictionary when creating the list and instead should always use `List<>`. 

If we need to force the order all the time, we can use IReadOnlyList<KeyValuePair<string, object>> instead but this will restrict the callers to use something like List<> all the time and cannot use IDictionary if they don't really care about the order. My take here is not restricting that and leave it to the caller to decide about the list order and how to create it.

Reply:
Thanks for clarify. After reading, that makes sense and you are right. In a way, before we could not maintain order. This will allow us to maintain order if given an ordered collection type.